test/test_spark.py

```diff
@@ -68,12 +70,24 @@ def __init__(self, *args, **kwargs):
         warnings.simplefilter('module')
 
     def run(self, result=None):
+        # Unit tests are usually ran via horovod, i.e. in a distributed way.
+        # That way, unit tests represent a single worker in a horovod cluster.
+        # The SparkTests work differently, they spawn their own cluster.
+        # Having workers span their own horovod cluster is a silly test setup.
+        # Therefore we skip all rank != 0 processes ...
         if int(os.getenv('OMPI_COMM_WORLD_RANK', 0)) != 0 or int(os.getenv('HOROVOD_RANK', 0)) != 0:
-            # Running in MPI or Gloo with rank > 0, ignore.
-            # Purposefully skip these silently
-            return
+            self.skipTest("Not testing for rank > 0, tests start their own MPI / Gloo setup")
 
-        super(SparkTests, self).run(result)
+        # ... and remove all HOROVOD_* environment variables.
+        # Pretend horovodrun is not there.
+        env = os.environ.copy()
```

Comment:
We should probably just change `test_spark.py` to run without mpirun/horovodrun, similar to what we do for `test_spark_torch.py`, etc.

Reply:
Good idea, thanks, have done this: https://github.com/horovod/horovod/pull/1839/commits/d956d623d14c10e3e0f3f0d9ac0091df393f9ea7

I have also added a test that compares `.buildkite/gen_pipeline.sh` with an expected yaml file, so we can track changes in the generated pipeline in git and PRs: https://github.com/horovod/horovod/pull/1839/commits/fccc745b442a0f93c1ecbed30b8c53b5d42a3f04